README.md

```diff
@@ -206,6 +206,32 @@ When using a GitHub personal access token, the username is arbitrary. GitHub doe
 
 If there are no credentials present in the URL and authentication is needed, the [Git credential helper](https://git-scm.com/doc/credential-helpers) will be queried.
 
+### HTTP authentication
+
+uv supports credentials over HTTP when querying package registries.
+
+Authentication can come from the following sources, in order of precedence:
+
+- The URL e.g. `https://<user>:<password>@<hostname>/...`
+- A [`netrc`](https://everything.curl.dev/usingcurl/netrc) configuration file
+- A [keyring](https://github.com/jaraco/keyring) provider (requires opt-in)
+
+If authentication is found for a single net location (scheme, host, and port), it will be cached for the duration
+of the command and used for other queries to that net location. Authentication is not cached across invocations of
+uv.
+
+Note `--keyring-provider subprocess` or `UV_KEYRING_PROVIDER=subprocess` must be provided to enable keyring-based
+authentication.
+
+Authentication may be used for hosts specified in the following options:
+
+- `index-url`
+- `extra-index-url`
+- `find-links`
```

Comment:
Do we also use authentication for `--requirement https://...` and `foo@https://...`?

Reply:
We do not do this yet

Reply:
We don't for requirement files. That's tracked in https://github.com/astral-sh/uv/issues/2357.

For `foo@https://...` I would expect that we do, yes.